sktime/transformations/series/date.py

```diff
@@ -154,14 +178,17 @@ def _transform(self, X, y=None):
         _check_feature_scope(self.feature_scope)
         _check_manual_selection(self.manual_selection, self.dummies)
 
-        Z = X.copy()
-
-        x_df = pd.DataFrame(index=Z.index)
-        if isinstance(x_df.index, pd.PeriodIndex):
-            x_df["date_sequence"] = Z.index.to_timestamp().astype("datetime64[ns]")
-        elif isinstance(x_df.index, pd.DatetimeIndex):
-            x_df["date_sequence"] = Z.index
-        elif not isinstance(x_df.index, pd.DatetimeIndex):
+        if isinstance(X.index, pd.MultiIndex):
+            time_index = X.index.get_level_values(-1)
+        else:
+            time_index = X.index
+
+        x_df = pd.DataFrame(index=X.index)
+        if isinstance(time_index, pd.PeriodIndex):
+            x_df["date_sequence"] = time_index.to_timestamp().astype("datetime64[ns]")
+        elif isinstance(time_index, pd.DatetimeIndex):
+            x_df["date_sequence"] = time_index
+        elif not isinstance(time_index, pd.DatetimeIndex):
```

Comment:
@danbartl - as we check whether `time_index` is of type `pd.DatetimeIndex` on line 189, should we change line 191 to be a simple `else`?

Reply:
I've gone ahead and made the change. Let me know if there are any issues with this!